src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8406,9 +8406,9 @@ private ImmutableArray<BoundExpression> GetDeconstructionRightParts(BoundExpress
 
                 // Analyze operator call properly (honoring [Disallow|Allow|Maybe|NotNull] attribute annotations) https://github.com/dotnet/roslyn/issues/32671
                 // https://github.com/dotnet/roslyn/issues/29961 Update conversion method based on operand type.
-                if (node.OperandConversion.IsUserDefined && node.OperandConversion.Method?.ParameterCount == 1)
+                if (node.OperandConversion is BoundConversion operandConversion && operandConversion.Conversion.IsUserDefined && operandConversion.Conversion.Method?.ParameterCount == 1)
```

Comment:
>node.OperandConversion is BoundConversion operandConversion [](http://example.com/codeflow?start=20&length=59)

Consider using `node.OperandConversion is BoundConversion { Conversion: var operandConversion }` since it looks like we only use the `BoundConversion.Conversion` property.